### PR TITLE
Introspection support

### PR DESCRIPTION
**Pull request by tvvignesh**
## Description

Added support for introspection json file

Fixes https://github.com/kamilkisiela/graphql-inspector/issues/1769

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

https://github.com/vignesh-test-org/gql-inspector/pull/1 shows the action working with introspection file

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/action/src/run.ts [hunk 1]
```diff
@@ -136,14 +136,29 @@ export async function run() {
 
   core.info('Got both sources');
 
-  const sources = {
-    old: new Source(oldFile, endpoint || `${schemaRef}:${schemaPath}`),
-    new: new Source(newFile, schemaPath),
-  };
+  let oldSchema:GraphQLSchema, newSchema:GraphQLSchema, sources, schemas;
```
Comment (kamilkisiela) [inline packages/action/src/run.ts:139]:
```suggestion
  let oldSchema: GraphQLSchema;
  let newSchema: GraphQLSchema;
  let sources: { new: Source; old: Source }; 
```

File: packages/action/src/run.ts [hunk 2]
```diff
@@ -136,14 +136,29 @@ export async function run() {
...
+    };
 
-  const schemas = {
-    old: produceSchema(sources.old),
-    new: produceSchema(sources.new),
+    oldSchema = produceSchema(sources.old);
+    newSchema = produceSchema(sources.new);
+  }
+
+  schemas = {
```
Comment (kamilkisiela) [inline packages/action/src/run.ts:159]:
```suggestion
  const schemas = {
```

File: packages/action/src/run.ts [hunk 3]
```diff
@@ -136,14 +136,31 @@ export async function run() {
...
 
-  const sources = {
-    old: new Source(oldFile, endpoint || `${schemaRef}:${schemaPath}`),
-    new: new Source(newFile, schemaPath),
-  };
+  let oldSchema: GraphQLSchema;
+  let newSchema: GraphQLSchema;
+  let sources: { new: Source; old: Source; }; 
+
+  if(extname(schemaPath.toLowerCase())===".json") {
```
Comment (kamilkisiela) [inline packages/action/src/run.ts:143]:
```suggestion
  if (extname(schemaPath.toLowerCase()) === ".json") {
```

File: packages/action/src/run.ts [hunk 4]
```diff
@@ -136,14 +136,31 @@ export async function run() {
...
-    old: new Source(oldFile, endpoint || `${schemaRef}:${schemaPath}`),
-    new: new Source(newFile, schemaPath),
-  };
+  let oldSchema: GraphQLSchema;
+  let newSchema: GraphQLSchema;
+  let sources: { new: Source; old: Source; }; 
+
+  if (extname(schemaPath.toLowerCase()) === ".json") {
+    oldSchema = buildClientSchema(JSON.parse(oldFile));
+    newSchema = buildClientSchema(JSON.parse(newFile));
```
Comment (AllexVeldman) [inline packages/action/src/run.ts:145]: Introspection query results are normally part of `{ "data" : { ... } }` in the query output making this raise
```
Error: Invalid or incomplete introspection result. Ensure that you are passing "data" property of introspection response and no "errors" was returned alongside: { data: { __schema: [Object] } }.
```

Consider passing `JSON.parse(oldFile).data` to `buildClientSchema` if it exists in the parsed file

Comment (kamilkisiela) [reply packages/action/src/run.ts:145]: @tvvignesh

Comment (tvvignesh) [reply packages/action/src/run.ts:145]: I will have a look at this today.

Comment (tvvignesh) [reply packages/action/src/run.ts:145]: @AllexVeldman @kamilkisiela Have raised a PR for the same here: https://github.com/kamilkisiela/graphql-inspector/pull/2059

**Pull request merged**